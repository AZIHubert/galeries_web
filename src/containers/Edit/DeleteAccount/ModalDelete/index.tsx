import { useFormik } from 'formik';
import * as React from 'react';
import {
  useSelector,
} from 'react-redux';

import Button from '#components/Button';
import Field from '#components/Field';
import Text from '#components/Text';

import { deleteAccountSchema } from '#helpers/schemas';

import { loadingSelector } from '#store/selectors';

const initialValues = {
  deleteAccountSentence: '',
  password: '',
  userNameOrEmail: '',
};

const ModalDelete = () => {
  const formik = useFormik({
    initialValues,
    onSubmit: () => {},
    validateOnBlur: true,
    validateOnChange: false,
    validationSchema: deleteAccountSchema,
  });
  const loading = useSelector(loadingSelector);
  return (
    <div>
      <Text
        color='danger'
        styles={{
          fontSize: 1.1,
          marginBottom: 30,
        }}
      >
        Are you sure you want to do this?
      </Text>
      <form onSubmit={formik.handleSubmit}>
        <Field
          disabled={loading}
          error={formik.errors.userNameOrEmail}
          fieldTestId='field'
          id='userNameOrEmail'
          label='user name or email'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          styles={{
            marginBottom: 6,
          }}
          stylesLaptopL={{
            marginBottom: 10,
          }}
          touched={formik.touched.userNameOrEmail}
          type='text'
          value={formik.values.userNameOrEmail}
        />
        <Field
          disabled={loading}
          error={formik.errors.deleteAccountSentence}
          fieldTestId='field'
          id='deleteAccountSentence'
          label='To verify, type "delete my account" below:'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          styles={{
            marginBottom: 6,
          }}
          stylesLaptopL={{
            marginBottom: 10,
          }}
          touched={formik.touched.deleteAccountSentence}
          type='text'
          value={formik.values.deleteAccountSentence}
        />
        <Field
          disabled={loading}
          error={formik.errors.password}
          fieldTestId='field'
          id='password'
          label='confirm your password'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          touched={formik.touched.password}
          type='password'
          value={formik.values.password}
        />
        <Button.Default
          danger
          disabled={loading}
          data-testid='submitButton'
          styles={{
            marginBottom: 10,
            marginTop: 20,
          }}
          stylesLaptopL={{
            marginTop: 24,
          }}
          type='submit'
          title='delete your account'
        />
      </form>
    </div>
  );
};

export default ModalDelete;
