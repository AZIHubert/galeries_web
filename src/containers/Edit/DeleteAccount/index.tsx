import * as React from 'react';
import {
  useSelector,
} from 'react-redux';

import Modal from '#components/Modal';

import ModalDelete from './ModalDelete';

import { loadingSelector } from '#store/selectors';

const DeleteAccount = () => {
  const loading = useSelector(loadingSelector);
  const [openModal, setOpenModal] = React.useState<boolean>(false);
  const handleClose = () => {
    if (!loading) {
      setOpenModal(false);
    }
  };
  return (
    <div>
      <p>
        delete your account
      </p>
      <p>
        Once you delete your account, there is no going back. Please be certain.
      </p>
      <button
        onClick={() => setOpenModal((nextState) => !nextState)}
      />
      <Modal.Portal
        modalTestId='modal'
        open={openModal}
        handleClose={handleClose}
      >
        <ModalDelete />
      </Modal.Portal>
    </div>
  );
};

export default DeleteAccount;
