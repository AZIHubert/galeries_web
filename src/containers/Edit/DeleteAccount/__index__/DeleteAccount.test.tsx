import * as React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {
  cleanup,
  fireEvent,
  render,
  screen,
} from '@testing-library/react';

import DeleteAccount from '../index';

describe('DeleteAccount', () => {
  beforeAll(() => {
    // @ts-ignore
    ReactDOM.createPortal = jest.fn((element) => element);
  });
  afterEach(() => {
    // @ts-ignore
    ReactDOM.createPortal.mockClear();
  });
  afterEach(cleanup);
  it('renders without crashing', () => {
    const tree = renderer.create(<DeleteAccount />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should not display ticket modal on mount', () => {
    const { queryByTestId } = render(<DeleteAccount />);
    const deleteModal = queryByTestId('deleteModal');
    expect(deleteModal).toBeNull();
  });
  it('open the ticket modal', () => {
    const { getByTestId } = render(<DeleteAccount />);
    const buttonDeleteModal = getByTestId('buttonDeleteModal');
    fireEvent.click(buttonDeleteModal);
    const deleteModal = screen.queryByTestId('deleteModal');
    expect(deleteModal).not.toBeNull();
  });
  it('should close the ticket modal', () => {
    const { getByTestId } = render(<DeleteAccount />);
    const buttonDeleteModal = getByTestId('buttonDeleteModal');
    fireEvent.click(buttonDeleteModal);
    fireEvent.click(buttonDeleteModal);
    const deleteModal = screen.queryByTestId('deleteModal');
    expect(deleteModal).toBeNull();
  });
});
