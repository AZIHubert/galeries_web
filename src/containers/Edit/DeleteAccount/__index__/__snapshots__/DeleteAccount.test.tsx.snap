// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteAccount renders without crashing 1`] = `
<div>
  <p>
    delete your account
  </p>
  <p>
    Once you delete your account, there is no going back. Please be certain.
  </p>
  <button
    data-testid="buttonDeleteModal"
    onClick={[Function]}
  />
</div>
`;
